articles/tutorials/new_search.md

```diff
@@ -0,0 +1,476 @@
+---
+title: Implement Grover's search algorithm in Q# - Quantum Development Kit
+description: Build a Q# project that demonstrates Grover's algorithm, one of the canonical quantum algorithms.
+author: geduardo
+ms.author: v-edsanc
+ms.date: 11/30/2020
+ms.topic: article
+uid: microsoft.quantum.quickstarts.new-search
+no-loc: ['Q#', '$$v']
+---
+
+# Tutorial: Implement Grover's search algorithm in Q\#
+
+In this tutorial you'll learn to implement Grover's algorithm in Q# to solve search based problems.
+
+Grover's algorithm is one of the most famous algorithms in quantum computing. The problem it solves is often referred to as "searching a database", but it's more accurate to think of it as a "search problem".
+
+> [!NOTE] 
+> This tutorial is intended for people who are already familiar with
+> Grover's algorithm that want to learn how to implement it in Q#. For a more
+> slow paced tutorial we recommend the Microsoft Learn module [Solve graph
+> coloring problems by using Grover's
+> search](https://docs.microsoft.com/learn/modules/solve-graph-coloring-problems-grovers-search/).
+
```

Comment:
I wonder if the MS Learn module is going to satisfy people who don't already know Grover's algorithm, since the module is more about its application, rather than an explanation about how Grover's algorithm is constructed? We can either add that in this tutorial or update the Grover's module or in the conceptual module. I'm also not sure, having learned the algorithm using circuits, if programmers find it easy to understand the algorithm without the aim of the circuit.

Reply:
The MS Learn module doesn't explain the implementation (that's the task of this tutorial), but it does explain why Grover's algorithm work (for example, with the rotation's diagrams).

About the circuits, I don't think we need them. You just need to have clear the steps of the algorithm. It could be useful tho a mathematical description of how the state of the register evolves during the algorithm.

Reply:
"It could be useful tho a mathematical description of how the state of the register evolves during the algorithm." Can you add this?

Reply:
Mm, I think that properly adding that here in this tutorial would make it too long and heavy for what's supposed to be a "quick implementation guide". But I also think that it could too mathematically dense for the MS Learn module.

What do you think about adding a conceptual article in the Concepts section (where we can use more "heavy maths") to explain properly Grover's algorithm? We can use it then as a reference for both, the MS Learn module and the implementation tutorial, for those users that want to dive deep into the math.

Reply:
Agreed. Adding it to the conceptual article makes sense.

Reply:
Ok! I'll write it up